src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/InstantiateCommand.cs

```diff
@@ -511,7 +511,7 @@ private static void WriteListCommandExample(InstantiateCommandArgs instantiateAr
                   Example
                       .For<NewCommand>(instantiateArgs.ParseResult)
                       .WithSubcommand<ListCommand>()
-                      .WithArgument(ListCommand.NameArgument, instantiateArgs.ShortName));
+                      .WithArgument(BaseListCommand.NameArgument, instantiateArgs.ShortName));
```

Comment:
Was this done automatically? Why is it ok?

Reply:
I agree with Nathan, I prefer the 'ListCommand' over 'BaseListCommand'... same for 'BaseUninstallCommand' vs 'UninstallCommand', etc... all of the other stuff looks awesome though :)

Reply:
So, the way the rule works is that it tries to apply the most *precise* way of accessing members. Here, `NameArgument` doesn't exist as a member on `ListCommand` directly. `NameArgument` is a static member in `BaseListCommand`. However, `ListCommand` derives of `BaseListCommand` and static members can be accessed transitively through it.

Honestly, asking about this situation convinced me *why* this rule should be done, as opposed to changing it back. The reason is that this shows the design choices being made and brings to light how things could be architected differently. The question that could be asked here instead is, "Why is InstantiateCommand accessing an argument from the BaseListCommand?" Other places in these CLI commands use a `SharedOptions` class for this purpose of keeping static options. Maybe creating a `SharedArguments` would be a cleaner design? That's kinda how I think about things. 🙃

Reply:
I can see your argument. I'm guessing the initial reason for why it accessed an argument from BaseListCommand is that it was there, and someone needed it, so they used it—more pragmatic than architectural.

One concern I'd have with switching to the BaseListCommand is that if the person who wrote this actually was thinking architecturally, perhaps there's a concept of a "NameArgument" on ListCommand that differs from the concept of a "NameArgument" on BaseListCommand in some critical way. I can't think of a reason here, but say there were an "UpdateCounter" method on both, but the ListCommand has an extra counter to update. ListCommand would then override that base method. We'd want ListCommand's version regardless of whether it had actually overridden it yet...so if it isn't overridden yet, VS might suggest a change like this, but then we wouldn't automatically pick up the override from ListCommand if it comes.

That was a little contorted; did it make sense?

Reply:
@Forgind 
> We'd want ListCommand's version regardless of whether it had actually overridden it yet...so if it isn't overridden yet, VS might suggest a change like this

It doesn't. The analyzer has contextual awareness of how the classes are constructed. There could be situations like those that exist in the codebase right now, but this PR wouldn't have changed those. From my basic understanding of the IL, they are essentially 2 different callsites. The fact they have the same name only applies to the user (aka us, the devs) as we'd need to add a `new` to override the name conflict. Roslyn is used to drive these analyzers so it has those smarts. Trust the tooling 😉 (until we make it **too** smart and it becomes Skynet, and now we fear for our mortal lives... 💥💀)

Reply:
Yeah; I agree this won't break anything. For the example I provided with UpdateCounter, the idea was that at the outset (i.e., now) ListCommand doesn't override BaseListCommand's implementation of UpdateCounter, but if it does so in the future, which should we use? I'd imagine we should use ListCommand's version, and I think this would make us use BaseListCommand's.

Reply:
In this context, there is no intrinsic relationship between the usage in this code and the base/non-base version of `ListCommand`. This code is the `dotnet new create` command (aka, `dotnet new <TEMPLATE>`) and the `Base/ListCommand` is from `dotnet new list`. Picking either one could make sense depending on the goal of the developer. Personally, I don't think there is any native bias between one or the other.

If there is a `new`ed version of `NameArgument`, the developer would need to be aware to investigate *all* callsites of the old one either way. One cannot assume the new version would work in every situation.